sphinx_gallery/tests/test_gen_rst.py

```diff
@@ -185,6 +185,17 @@ def test_ipy_notebook():
         f.flush()
         assert_equal(json.load(f), example_nb.work_notebook)
 
+def test_thumbnail_number():
+    # which plot to show as the thumbnail image
+    for test_str in ['# sphinx_gallery_thumbnail_number: 2',
+                     '# sphinx_gallery_thumbnail_number:2',
+                     '#sphinx_gallery_thumbnail_number: 2']:
+        with tempfile.NamedTemporaryFile('w') as f:
+            f.write('\n'.join(['"Docstring"',
+                               test_str]))
+            f.flush()
+            _,cfg = sg.split_code_and_text_blocks(f.name)
+        assert cfg['thumbnail_number'] == 2
```

Comment:
The error message is slightly better with `assert_equal(cfg['thumbnail_number'], 2)` from `nose.tools`.